kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java

```diff
@@ -1223,20 +1225,87 @@ public void testFindTaskEvents() throws Exception {
             events = taskClient.findTaskEvents(CONTAINER_ID, taskInstance.getId(), 0, 10);
             assertNotNull(events);
             assertEquals(2, events.size());
+            assertTaskEventInstance(expectedTaskEventInstance, events.get(0));
+            expectedTaskEventInstance.setType(TaskEvent.TaskEventType.STARTED.toString());
+            expectedTaskEventInstance.setUserId(USER_YODA);
+            assertTaskEventInstance(expectedTaskEventInstance, events.get(1));
         } finally {
             processClient.abortProcessInstance(CONTAINER_ID, processInstanceId);
         }
     }
 
+    @Test
+    public void testFindTaskEventsForNotExistingTask() {
+        try {
+            taskClient.findTaskEvents(CONTAINER_ID, -9999l, 0, 10);
+            fail("KieServicesException should be thrown complaining about task not found.");
+        } catch (KieServicesException e) {
+            KieServerAssert.assertResultContainsString(e.getMessage(), "No task found with id");
+        }
+    }
+
+    @Test
+    public void testFindTaskEventsSortedBy() throws Exception {
+        Map<String, Object> parameters = new HashMap<String, Object>();
+        parameters.put("stringData", "waiting for signal");
+        parameters.put("personData", createPersonInstance(USER_JOHN));
+
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK, parameters);
+
+        try {
+
+            List<TaskSummary> tasks = taskClient.findTasksByStatusByProcessInstanceId(processInstanceId, null, 0, 10);
+            assertNotNull(tasks);
+            assertEquals(1, tasks.size());
+
+            TaskSummary taskInstance = tasks.get(0);
+
+            taskClient.startTask(CONTAINER_ID, taskInstance.getId(), USER_YODA);
+            taskClient.stopTask(CONTAINER_ID, taskInstance.getId(), USER_YODA);
+
+            List<TaskEventInstance> events = taskClient.findTaskEvents(CONTAINER_ID, taskInstance.getId(), 0, 10, SORT_BY_TASK_EVENTS_TYPE, true);
+            assertEquals(TaskEvent.TaskEventType.ADDED.toString(), events.get(0).getType());
+            assertEquals(TaskEvent.TaskEventType.STARTED.toString(), events.get(1).getType());
+            assertEquals(TaskEvent.TaskEventType.STOPPED.toString(), events.get(2).getType());
+
+            events = taskClient.findTaskEvents(CONTAINER_ID, taskInstance.getId(), 0, 10, SORT_BY_TASK_EVENTS_TYPE, false);
+            assertEquals(TaskEvent.TaskEventType.STOPPED.toString(), events.get(0).getType());
+            assertEquals(TaskEvent.TaskEventType.STARTED.toString(), events.get(1).getType());
+            assertEquals(TaskEvent.TaskEventType.ADDED.toString(), events.get(2).getType());
+        } finally {
+            processClient.abortProcessInstance(CONTAINER_ID, processInstanceId);
+        }
+    }
+
+    private void assertTaskEventInstance(TaskEventInstance expected, TaskEventInstance actual) {
```

Comment:
It is a good idea to use AssertJ but I think that would be better to use AssertJ for whole classes (or kie server jBPM integration tests module). If we do changes only for this PR, then we will have a few tests that use AssertJ and the other tests use JUnit and I am afraid that this could be confusing for others.